tests/integration/cluster_fixture.py

```diff
@@ -0,0 +1,226 @@
+        """
+        Creates a new topic with this cluster.
+
+        :param str name: topic name
+        :param dict topic_options: additions/overrides to topic configuration.
+        :returns: The topic's name
+        :rtype: str
+        """
+        if self._admin is None:
+            self._admin = AdminClient(self.client_conf())
+
+        future_topic = self._admin.create_topics([NewTopic(name, **self._topic_conf(topic_options))])
+
+        future_topic.get(name).result()
+        return name
+
+    def seed_topic(self, topic, value_source=None, key_source=None, header_source=None):
+        """
+        Populates a topic with data asynchronously.
+
+        If there are fewer keys or headers than values the index will be wrapped and reused with
+        each subsequent produce request.
+
+        :param str topic: topic to produce test data to.
+        :param list value_source: list or other iterable containing message key data to be produced.
+        :param list key_source: list or other iterable containing message value data to be produced.
+        :param list(dict) header_source: headers to attach to test data.
+        """
+
+        if self._producer is None:
+            self._producer = Producer(self.client_conf({'linger.ms': 500}))
+
+        if value_source is None:
+            value_source = ['test-data{}'.format(i) for i in range(0, 100)]
+
+        if key_source is None:
+            key_source = [None]
+
+        produce_thread = threading.Thread(target=_produce,
+                                          args=(self._producer, topic, value_source,
+                                                key_source, header_source))
+
+        produce_thread.daemon = True
+        produce_thread.start()
+
+    @property
+    def schema_registry(self):
+        return self._cluster.sr.get('url')
+
+
+def _produce(producer, topic, value_source, key_source, header_source=None):
+    num_messages = len(value_source)
+    num_keys = len(key_source)
+    num_headers = len(header_source) if header_source else 0
+
+    print('# producing {} messages to topic {}'.format(num_messages, topic))
+    for i in range(0, num_messages):
+        while True:
+            try:
+                if header_source is not None:
+                    producer.produce(topic,
+                                     value=value_source[i],
+                                     key=key_source[i % num_keys],
+                                     headers=header_source[i % (num_headers + 1)])
+                else:
+                    producer.produce(topic,
+                                     value=value_source[i],
+                                     key=key_source[i % num_keys])
+                break
+            except BufferError:
+                producer.poll(0.1)
+            continue
+        producer.poll(0.0)
+
+    producer.flush(10.0)
+
+    print('# finished producing {} messages to topic {}'.format(num_messages, topic))
```

Comment:
Since you are not checking the return value from flush(with a timeout) you dont know if all messages were sent.
Suggest you fail if flush() returns non-zero.

Reply:
how about we just block for -1

Reply:
Yes, that's the recommended way to call flush(), to rely on the message.timeout.ms